src/app/route/ProductPage/ProductPage.component.js

```diff
@@ -172,24 +172,36 @@ class ProductPage extends Component {
 
     /**
      * Update query params without adding to history, set configurableVariantIndex
-     * @param {Number} variant
+     * @param {string} key
+     * @param {number|string} value
      */
-    updateUrl(variant) {
-        const { configurableVariantIndex } = this.state;
+    updateUrl(key, value) {
+        const { product: { variants, configurable_options } } = this.props;
+        const { configurableVariantIndex, parameters: oldParameters } = this.state;
 
-        if (configurableVariantIndex !== variant) updateQueryParamWithoutHistory('variant', variant);
+        const parameters = {
+            ...oldParameters,
+            [key]: `${value}`
```

Comment:
Is this `toString()` conversion? I would prefer `.toString()` or I would try `value` directly, as it is URL => logically a string. Please check.